Plugins/Json/include/Acts/Plugins/Json/JsonHierarchicalObjectConverter.hpp

```diff
@@ -0,0 +1,81 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2019 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/GeometryID.hpp"
+#include "Acts/Geometry/HierarchicalGeometryContainer.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include <map>
+
+namespace Acts {
+
+/// Convert Hierarchical Object Container to Json file and vice-versa
+template <typename object_t>
+struct JsonHierarchicalObjectConverter {
+ public:
+  /// The object key
+  std::string datakey = "Object";
+
+  /// The loglevel
+  Acts::Logging::Level logLevel = Acts::Logging::Level::INFO;
+
+  /// Convert json map to Hierarchical Object Container
+  ///
+  /// @param map The indexed Hierarchical Object in json format
+  /// @param fromJson Function that return Hierarchical Object from json
+  HierarchicalGeometryContainer<object_t> jsonToHierarchicalContainer(
+      const nlohmann::json& map,
+      std::function<object_t(const nlohmann::json&, const Acts::GeometryID&)>
```

Comment:
This converts a key/value pair with the geometry identifier as the key. The canonical order of the arguments would therefore be `(GeometryID, const nlohman::json&)`. The geometry identifier can also be given by value.